<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Protein Sequence Diversity Dynamics Analyser for Viruses (DiMA) &mdash; DiMA Help Page .. documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> DiMA Help Page
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Protein Sequence Diversity Dynamics Analyser for Viruses (DiMA)</a></li>
<li><a class="reference internal" href="#about">About</a><ul>
<li><a class="reference internal" href="#defining-diversity-motifs">Defining diversity motifs</a></li>
</ul>
</li>
<li><a class="reference internal" href="#browser-compatibility">Browser compatibility</a><ul>
<li><a class="reference internal" href="#accesibility">Accesibility</a></li>
</ul>
</li>
<li><a class="reference internal" href="#workflow">Workflow</a></li>
<li><a class="reference internal" href="#input-file-and-parameters">Input file and parameters</a></li>
<li><a class="reference internal" href="#how-to-interpret-the-results">How to interpret the results</a></li>
<li><a class="reference internal" href="#faqs">FAQs</a></li>
<li><a class="reference internal" href="#support">Support</a><ul>
<li><a class="reference internal" href="#team">Team</a></li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">DiMA Help Page</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Protein Sequence Diversity Dynamics Analyser for Viruses (DiMA)</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/myfile.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <ul class="simple">
<li><p><a class="reference internal" href="#section-one"><span class="std std-ref">About</span></a></p></li>
<li><p><a class="reference internal" href="#section-two"><span class="std std-ref">Browser compatibility</span></a></p></li>
<li><p><a class="reference internal" href="#section-three"><span class="std std-ref">Workflow of DiMA</span></a></p></li>
<li><p><a class="reference internal" href="#section-four"><span class="std std-ref">Input file and parameters</span></a></p></li>
<li><p><a class="reference internal" href="#section-five"><span class="std std-ref">How to interpret the results</span></a></p></li>
<li><p><a class="reference internal" href="#section-six"><span class="std std-ref">FAQs</span></a></p></li>
<li><p><a class="reference internal" href="#section-seven"><span class="std std-ref">Support</span></a></p></li>
</ul>
<section class="tex2jax_ignore mathjax_ignore" id="protein-sequence-diversity-dynamics-analyser-for-viruses-dima">
<h1>Protein Sequence Diversity Dynamics Analyser for Viruses (DiMA)<a class="headerlink" href="#protein-sequence-diversity-dynamics-analyser-for-viruses-dima" title="Permalink to this headline"></a></h1>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="about">
<span id="section-one"></span><h1>About<a class="headerlink" href="#about" title="Permalink to this headline"></a></h1>
<p>Viral infectious diseases are a major public health threat. Protein sequence diversity is one of the major challenges in the design of diagnostic, prophylactic and therapeutic interventions against viruses. The diversity can be an outcome of a combination of underlying evolutionary processes (mutation, recombination, and assortment). A continuing goal is a greater understanding of viral proteome sequence diversity, the dynamics of substitutions, and effective strategies to overcome the diversity for drug or vaccine design.</p>
<p>Herein, we present Diversity Motif Analyser (DiMA), a tool designed to facilitate the quantification and dissection of viral protein sequence diversity dynamics. DiMA provides a quantitative measure of sequence diversity by use of Shannon’s entropy (PMID: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/18698358/">18698358</a>), applied via a user-defined <em>k-mer</em> sliding window to a protein alignment. Additionally, DiMA further interrogates the diversity by dissecting the entropy at each aligned <em>k-mer</em> position to various diversity motifs (PMIDs: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/32518710/">32518710</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/23593157/">23593157</a>), based on the incidence of distinct <em>k-mer</em> sequences at the position. At a given position, index is the predominant sequence and all other distinct <em>k-mer</em>s are referred to as total variants to the index, sub-classified into major variant (the most common variant), minor variants (comprising of <em>k-mer</em>s with incidence lower than major and higher than unique), and unique variants (<em>k-mer</em>s seen only once in the alignment). Moreover, the description line of the sequences in the input alignment can be enriched for inclusion of meta-data as part of the analysis, such as spatio-temporal information, among others. DiMA outputs a JSON file that provides multiple facets of sequence diversity: sequence name, <em>k-mer</em> position, entropy, distinct <em>k-mer</em>s at the position, and their incidence, motif classification and metadata (if available). DiMA enables comparative sequence diversity dynamics analyses, within and between proteins of a virus species, and proteomes of different species.</p>
<p>DiMA is an outcome of many years of viral studies on several different species since 2007. These studies have been published in peer-reviewed journals (PMIDs: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/17434154/">17434154</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/18030326/">18030326</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/18698358/">18698358</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/19401763/">19401763</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/21471731/">21471731</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/22573867/">22573867</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/23593157/">23593157</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/26680743/">26680743</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29322922/">29322922</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29363421/">29363421</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/31874646/">31874646</a>, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/34068495/">34068495</a> and <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/32518710/">32518710</a> and international conferences. Besides being available as a webserver, it can also be downloaded as a standalone client tool (<a class="reference external" href="https://github.com/PU-SDS/DiMA">https://github.com/PU-SDS/DiMA</a>), particularly for big data analyses.</p>
<p>DiMA webserver has been under development since March 2020. It has been extensively tested with 18 protein datasets (9 structural and 9 non-structural) from six viral species. External validation of our tool has been performed by three individuals, with a total of 36 protein datasets (18 structural and 18 non-structural), originating from three viral species.</p>
<section id="defining-diversity-motifs">
<h2>Defining diversity motifs<a class="headerlink" href="#defining-diversity-motifs" title="Permalink to this headline"></a></h2>
<p>For a given protein alignment, all sequences at each of the aligned <em>k-mer</em> positions are quantified for distinct sequences and ranked-classified into diversity motifs based on their incidences, as described in Hu et al. (2013) (Supplementary Figure 1, see extract below) (PMID: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/23593157/">23593157</a>).</p>
<a class="bg-primary reference internal image-reference" href="_images/motifs.jpg"><img alt="diversitymotifs" class="bg-primary align-center" src="_images/motifs.jpg" style="width: 300px;" /></a>
<p><br><br/>
<strong>Figure 1: Definitions of diversity motifs.</strong> The ‘‘Index’’ nonamer is the most prevalent sequence, present in 12 of the 20 isolates. The ‘‘Major’’ variant is the most common variant of the index (5/20). ‘‘Minor’’ variants are multiple different repeated sequences, each with incidences less than the major variant. ‘‘Unique’’ variants are those represented by a single aligned sequence. Distinct variant sequences at a given nonamer position are the different sequence at the position; in this example one of major, two of minor, and three of unique.</p>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="browser-compatibility">
<span id="section-two"></span><h1>Browser compatibility<a class="headerlink" href="#browser-compatibility" title="Permalink to this headline"></a></h1>
<p><img alt="browserc" src="_images/browserc.png" /></p>
<section id="accesibility">
<h2>Accesibility<a class="headerlink" href="#accesibility" title="Permalink to this headline"></a></h2>
<p>The webserver is publicly available at:
https://dima.bioinfo.perdanauniversity.edu.my</p>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="workflow">
<span id="section-three"></span><h1>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline"></a></h1>
<p><img alt="workflow" src="_images/workflow.jpg" />
<strong>Figure 1</strong><br />
<strong>Input:</strong> Viral protein sequences, typically obtained from publicly available databases (NCBI virus and GISAID, among others), aligned and submitted to DiMA in aligned FASTA (<em>.afa)</em> format.<br />
<strong>Process:</strong> DiMA provides a quantitative measure of sequence diversity by use of Shannon’s entropy, applied via a user-defined <em>k-mer</em> sliding window. Further, the entropy value is corrected for sample size bias by applying a statistical adjustment (Lipinski’s rule). Additionally, DiMA further interrogates the diversity by dissecting the entropy value at each <em>k-mer</em> position to various distinct <em>k-mer</em> sequences that are classified into diversity motifs (index, major, minor and unique; see Section 3 for the definition of the diversity motifs) based on their incidence.<br />
<strong>Output:</strong> The entropy values, diversity motifs, and each of the <em>k-mer</em> corresponding metadata is plotted to provide a panoramic overview of the protein sequence diversity.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="input-file-and-parameters">
<span id="section-four"></span><h1>Input file and parameters<a class="headerlink" href="#input-file-and-parameters" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><strong>Input file</strong><br />
DiMA only uses multiple sequence alignment (protein sequences; DNA should also work) in (aligned) FASTA (<em>.afa</em> or <em>.fas</em>) format. Any existing, published alignment tool can be used to produce the MSA, such as MAFFT or MUSCLE, as long as the aligned sequences are provided to DiMA as input in (aligned) FASTA format.</p></li>
<li><p><strong>Parameters</strong></p>
<ul>
<li><p><strong>Viral sequence name</strong><br />
Name of protein to be analysed (such as Nucleoprotein, Matrix protein, or Polymerase protein).</p></li>
<li><p><strong>Low support threshold</strong><br />
The support is defined as the number of sequences at a given <em>k-mer</em> position that do not harbor gap and unknown and/or ambiguous residues. Positions below a statistical support of 30 sequences (default) are defined as of low support. The user has the flexibility to set the threshold for low support.</p></li>
<li><p><strong>Kmer length</strong><br />
Select a <em>k-mer</em> window size that is appropriate for the analysis.\
While the minimum applicable size is 3, the maximum can equal to the alignment length of the uploaded input file. By default, DiMA uses a window size of nine (9; nonamer; 9-mer) to evaluate the viral diversity with respect to cellular immune response.</p></li>
<li><p><strong>Header format to include metadata</strong><br />
This optional functionality allows annotation of the distinct sequences at each <em>k-mer</em> position  with respective cognate sequence metadata, such as collection date, country of isolation , isolation host. Simply, it parses the information on the sequence header (definition/description line).</p></li>
</ul>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Example of a definition line: &gt;ATY74257.1 |2017-03-02|China: Kunming|Homo sapiens</p>
</div>
<p>Because the format of metadata varies between databases, DiMA has relied on the format of <a class="reference external" href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/">NCBI Virus</a>.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="how-to-interpret-the-results">
<span id="section-five"></span><h1>How to interpret the results<a class="headerlink" href="#how-to-interpret-the-results" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><strong>DiMA Output</strong></p></li>
</ul>
<p><img alt="sections" src="_images/sectionss.png" /></p>
<ul class="simple">
<li><p><strong>Section 1 (Summary statistics)</strong></p></li>
</ul>
<p>A summary of information about the query: request ID, submission parameters, the total number of sequences in the alignment, and the number of low support positions.</p>
<p><img alt="section1" src="_images/section1.png" /></p>
<ul class="simple">
<li><p><strong>Section 2 (Proteome diversity)</strong></p></li>
</ul>
<p>Entropy values indicate the level of variability at the corresponding <em>k-mer</em> positions, with zero representing completely conserved positions. Plots provide a holistic view of the diversity and are responsive and interactive (one can easily hover and see the approximate entropy value of the hovered position).</p>
<p><img alt="section2" src="_images/section2.png" /></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For a benchmark, the peak absolute entropy of 9.2 and total variants of 98% were observed for HIV-1 clade B <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0059994">(Hu et al., 2013)</a>.</p>
</div>
<p>The methodology for calculation of Shannon’s entropy at each <em>k-mer</em> position is as per <a class="reference external" href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0000272">Khan et al., (2008)</a>.</p>
<ul class="simple">
<li><p><strong>Section 3 (Diversity motifs)</strong></p></li>
</ul>
<p>All sequences at each of the <em>k-mer</em> positions in the protein alignments were quantified and quantified for distinct sequences and ranked-classified into diversity motifs based on their incidences, as explained above under the <a class="reference internal" href="#section-one"><span class="std std-ref">About</span></a> section.</p>
<p>Users can select a position from the “SELECTED POSITION” box, in the upper right corner to browse the motif distribution of the position.</p>
<a class="bg-primary reference internal image-reference" href="_images/section3.png"><img alt="section3" class="bg-primary align-center" src="_images/section3.png" style="width: 400px;" /></a>
<p><br><br/></p>
<ul class="simple">
<li><p><strong>Section 4 (Metadata)</strong></p></li>
</ul>
<p>If the header format is provided in the analysis parameters (as described in the above <a class="reference internal" href="#section-four"><span class="std std-ref">Parameters</span></a>), DiMA will make a pie chart for each type of the metadata.</p>
<p>The user should select a specific <em>k-mer</em> from the selected position for the metadata to appear. By default, the first peptide will be selected. In the example below, the index sequence is selected and host species distribution is shown in the plot.</p>
<p><img alt="section4" src="_images/section4.png" /></p>
<ul class="simple">
<li><p><strong>Download</strong></p></li>
</ul>
<p>Users can download the raw results file in JSON format from the most right bottom icon.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="faqs">
<span id="section-six"></span><h1>FAQs<a class="headerlink" href="#faqs" title="Permalink to this headline"></a></h1>
<ol class="arabic simple">
<li><p>How to cite?</p></li>
</ol>
<p>For now: <strong>Shan et al., https://dima.bioinfo.perdanauniversity.edu.my/</strong></p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="support">
<span id="section-seven"></span><h1>Support<a class="headerlink" href="#support" title="Permalink to this headline"></a></h1>
<p>Please don’t hesitate to reach out to the developers for your questions, comments, or other feedback through mailing <em>bioinfo&#64;perdanauniversity.edu.my</em></p>
<section id="team">
<h2>Team<a class="headerlink" href="#team" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Shan Tharanga</p></li>
<li><p>Yongli Hu</p></li>
<li><p>Olivo Miotto</p></li>
<li><p>Eyyüb Selim Ünlü</p></li>
<li><p>Muhammet A. Çelik</p></li>
<li><p>Muhammed Miran Öncel</p></li>
<li><p>Hilal Hekimoğlu</p></li>
<li><p>Muhammad Farhan Sjaugi</p></li>
<li><p>Mohammad Asif Khan</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Authors.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>