<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>About &mdash; DiMA Help Page .. documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> DiMA Help Page
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">About</a></li>
<li><a class="reference internal" href="#browser-compatibility">Browser compatibility</a><ul>
<li><a class="reference internal" href="#accesibility">Accesibility</a></li>
</ul>
</li>
<li><a class="reference internal" href="#workflow">Workflow</a></li>
<li><a class="reference internal" href="#input-file-and-parameters">Input file and parameters</a></li>
<li><a class="reference internal" href="#how-to-interpret-the-results">How to interpret the results</a></li>
<li><a class="reference internal" href="#faqs">FAQs</a></li>
<li><a class="reference internal" href="#support">Support</a><ul>
<li><a class="reference internal" href="#team">Team</a></li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">DiMA Help Page</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>About</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/myfile.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <ul class="simple">
<li><p><a class="reference internal" href="#section-one"><span class="std std-ref">About</span></a></p></li>
<li><p><a class="reference internal" href="#section-two"><span class="std std-ref">Browser compatibility</span></a></p></li>
<li><p><a class="reference internal" href="#section-three"><span class="std std-ref">Workflow of DiMA</span></a></p></li>
<li><p><a class="reference internal" href="#section-four"><span class="std std-ref">Input file and parameters</span></a></p></li>
<li><p><a class="reference internal" href="#section-five"><span class="std std-ref">How to interpret the results</span></a></p></li>
<li><p><a class="reference internal" href="#section-six"><span class="std std-ref">FAQs</span></a></p></li>
<li><p><a class="reference internal" href="#section-seven"><span class="std std-ref">Support</span></a></p></li>
</ul>
<section class="tex2jax_ignore mathjax_ignore" id="about">
<span id="section-one"></span><h1>About<a class="headerlink" href="#about" title="Permalink to this headline"></a></h1>
<p>Viral infectious diseases are a major public health threat. Protein sequence diversity is one of the major challenges in the design of diagnostic, prophylactic and therapeutic interventions against viruses. Herein, we present DiMA, a tool designed to facilitate the quantification and dissection of viral protein sequence diversity dynamics. DiMA provides a quantitative measure of sequence diversity by use of Shannon’s entropy, applied via a user-defined k-mer sliding window to a protein alignment. Additionally, DiMA further interrogates the diversity by dissecting the entropy at each aligned k-mer position to various diversity motifs, based on the incidence of distinct k-mer sequences at the position. At a given position, index is the predominant sequence and all other distinct k-mers are referred to as total variants to the index, sub-classified into major variant (the common variant), minor variants (comprising of k-mers with incidence lower than major and higher than unique), and unique variants (k-mers seen once in the alignment). Moreover, the description line of the sequences in the input alignment can be enriched for inclusion of meta-data as part of the analysis, such as spatio-temporal information, among others. DiMA outputs a JSON file that provides multiple facets of sequence diversity: sequence name, k-mer position, entropy, distinct k-mers at the position, and their incidence, motif classification and metadata (if available). DiMA enables comparative sequence diversity dynamics analyses, within and between proteins of a virus species, and proteomes of different species.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="browser-compatibility">
<span id="section-two"></span><h1>Browser compatibility<a class="headerlink" href="#browser-compatibility" title="Permalink to this headline"></a></h1>
<p><img alt="browserc" src="_images/browserc.png" /></p>
<section id="accesibility">
<h2>Accesibility<a class="headerlink" href="#accesibility" title="Permalink to this headline"></a></h2>
<p>The webserver is publicly available at:
https://dima.bioinfo.perdanauniversity.edu.my</p>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="workflow">
<span id="section-three"></span><h1>Workflow<a class="headerlink" href="#workflow" title="Permalink to this headline"></a></h1>
<p><img alt="workflow" src="_images/workflow.jpg" />
<strong>Figure 1</strong><br />
<strong>Input:</strong> Viral protein sequences are obtained from publicly available databases (NCBI virus, GISAID, etc.), aligned and submitted to DiMA in FASTA format.<br />
<strong>Process:</strong> DiMA provides a quantitative measure of sequence diversity by use of Shannon’s entropy, applied via a user-defined <em>k-mer</em> sliding window. Further, the entropy value is corrected for sample size bias by applying a statistical adjustment. Additionally, DiMA further interrogates the diversity by dissecting the entropy value at each k-mer position to various diversity motifs. The distinct <em>k-mer</em> sequences at each position are classified into the following motifs based on their incidence.<br />
<strong>Output:</strong> The entropy values, diversity motifs, and each of the <em>k-mer</em> corresponding metadata is plotted to provide a panoramic overview of the protein sequence diversity.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="input-file-and-parameters">
<span id="section-four"></span><h1>Input file and parameters<a class="headerlink" href="#input-file-and-parameters" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><strong>Input file</strong>
DiMA only uses multiple sequence alignment (DNA or protein sequences) in FASTA format. Any alignment tool can be used to produce the MSA, such as MAFFT or MUSCLE as long as the input sequences provided to DiMA are in FASTA format.</p></li>
<li><p><strong>Parameters</strong></p>
<ul>
<li><p><strong>Viral sequence name</strong><br />
Name your protein analysis (such as Nucleoprotein, Matrix protein, Polymerase protein etc.)</p></li>
<li><p><strong>Low support threshold</strong>\</p></li>
</ul>
</li>
</ul>
<p>Corrected entropy is provided by a sample size bias correction using Lipinski’s rule. The support is defined as the number of sequences at a given k-mer position that does not harbor gap and unknown or ambiguous residues.</p>
<p>The support for each position is 30 by default. The user has the flexibility to set the value.</p>
<ul class="simple">
<li><p><strong>Kmer length</strong>
Select <em>k-mer</em> window size appropriate for the analysis.\
While the minimum applicable size is 3, the maximum is equal to the total number position in the uploaded alignment. By default, DiMA uses a window size of nine (9; nonamer) to evaluate the viral diversity with respect to cellular immune response.</p></li>
<li><p><strong>Header format to include metadata</strong>
This optional functionality labels k-mers with corresponding metadata such as collection date, country origin, isolation host of the sequence. Simply, it parses the information on the sequence header (definition line).\</p></li>
</ul>
<p><img alt="headerformat" src="_images/headerformat.jpg" /></p>
<p>Because the format of metadata varies between databases, DiMA has relied on the format of NCBI Virus.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="how-to-interpret-the-results">
<span id="section-five"></span><h1>How to interpret the results<a class="headerlink" href="#how-to-interpret-the-results" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><strong>DiMA Output</strong></p></li>
</ul>
<p><img alt="sections" src="_images/sectionss.png" /></p>
<ul class="simple">
<li><p><strong>Section 1 (Summary statistics)</strong></p></li>
</ul>
<p>A summary of information about the query: request ID, submission parameters, the total number of sequences in the alignment, and the number of low support positions.</p>
<p><img alt="section1" src="_images/section1.png" /></p>
<ul class="simple">
<li><p><strong>Section 2 (Proteome diversity)</strong></p></li>
</ul>
<p>Entropy values indicate the level of variability at the corresponding k-mer positions, with zero representing completely conserved positions. Plots provide a holistic view of the diversity and are responsive and dynamic; one can easily hover and see the approximate entropy value of the hovered position.</p>
<p><img alt="section2" src="_images/section2.png" /></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For a benchmark, the peak absolute entropy of 9.2 and total variants of 98% were observed for HIV-1 clade B <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0059994">(Hu et al., 2013)</a>.</p>
</div>
<ul class="simple">
<li><p><strong>Section 3 (Diversity motifs)</strong></p></li>
</ul>
<p>All sequences at each of the <em>k-mer</em> positions in the protein alignments were quantified and ranked based on the incidences of distinct sequences (as part of diversity motifs) present at the position, as described in <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0059994">Hu et al. (2013)</a>. See in edge below for the definition of diversity motifs.</p>
<a class="bg-primary reference internal image-reference" href="_images/diversitymotifs.jpg"><img alt="diversitymotifs" class="bg-primary align-center" src="_images/diversitymotifs.jpg" style="width: 300px;" /></a>
<p><br><br/>
Users can select a position from the “SELECTED POSITION” box located in the upper right corner.
<br><br/></p>
<a class="bg-primary reference internal image-reference" href="_images/section3.png"><img alt="section3" class="bg-primary align-center" src="_images/section3.png" style="width: 400px;" /></a>
<p><br><br/></p>
<ul class="simple">
<li><p><strong>Section 4 (Metadata)</strong></p></li>
</ul>
<p>If the header format is provided in the analysis parameters, DiMA will make a pie chart for each type of the metadata.</p>
<p>The user should select a specific k-mer from the selected position for the metadata to appear. By default, the first peptide will be selected. In the example below, the index sequence is selected and host species distribution is shown in the plot.</p>
<p><img alt="section4" src="_images/section4.png" /></p>
<ul class="simple">
<li><p><strong>Download</strong></p></li>
</ul>
<p>Download the raw result file in JSON format from the most right bottom icon.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="faqs">
<span id="section-six"></span><h1>FAQs<a class="headerlink" href="#faqs" title="Permalink to this headline"></a></h1>
<ol class="arabic simple">
<li><p>How to cite?</p></li>
</ol>
<p>Shan et al., https://dima.bioinfo.perdanauniversity.edu.my/</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="support">
<span id="section-seven"></span><h1>Support<a class="headerlink" href="#support" title="Permalink to this headline"></a></h1>
<p>Please don’t hesitate to reach out to the developers for your questions, comments, or other feedback through mailing <em>bioinfo&#64;perdanauniversity.edu.my</em></p>
<section id="team">
<h2>Team<a class="headerlink" href="#team" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Shan Tharanga</p></li>
<li><p>Yongli Hu</p></li>
<li><p>Olivo Miotto</p></li>
<li><p>Eyyüb Selim Ünlü</p></li>
<li><p>Muhammet A. Çelik</p></li>
<li><p>Muhammed Miran Öncel</p></li>
<li><p>Hilal Hekimoğlu</p></li>
<li><p>Muhammad Farhan Sjaugi</p></li>
<li><p>Mohammad Asif Khan</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Authors.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>